src/core/ngx_connection.c

```diff
@@ -78,6 +78,7 @@ ngx_create_listening(ngx_conf_t *cf, struct sockaddr *sockaddr,
 
     ls->fd = (ngx_socket_t) -1;
     ls->type = SOCK_STREAM;
+    ls->protocol = 0;
```

Comment:
check how `ls` is initialised: maybe they do a calloc, and they only have to set non 0 values.

Reply:
correct, there is a memzero() call for ls. removing the line ^^